assembl/static2/js/app/components/administration/voteSession/tokensForm.jsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { I18n } from 'react-redux-i18n';
+import { Checkbox, DropdownButton, MenuItem } from 'react-bootstrap';
+import FormControlWithLabel from '../../common/formControlWithLabel';
+import { getEntryValueForLocale } from '../../../utils/i18n';
+import Helper from '../../common/helper';
+import TokenTypeForm from './tokenTypeForm';
+import {
+  updateTokenVoteInstructions,
+  createTokenVoteType,
+  deleteTokenVoteType,
+  updateTokenVoteExclusive
+} from '../../../actions/adminActions/voteSession';
+
+type TokensFormProps = {
+  instructions: string,
+  exclusive: boolean,
+  tokenTypeNumber: number,
+  tokenTypes: Object,
+  editLocale: string,
+  handleInstructionsChange: Function,
+  handleTokenVoteTypeNumberChange: Function,
+  handleExclusiveCheckboxChange: Function
+};
+
+const DumbTokensForm = ({
+  instructions,
+  exclusive,
+  tokenTypeNumber,
+  tokenTypes,
+  editLocale,
+  handleInstructionsChange,
+  handleTokenVoteTypeNumberChange,
+  handleExclusiveCheckboxChange
+}: TokensFormProps) => (
+  <div className="token-vote-form">
+    <form>
+      <div className="flex">
+        <Checkbox
+          checked={exclusive}
+          onChange={() => {
+            handleExclusiveCheckboxChange(exclusive);
+          }}
+        >
+          <Helper
+            label={I18n.t('administration.exclusive')}
+            helperText={I18n.t('administration.helpers.exclusive')}
+            classname="inline"
+            helperTextClassName="helper-text-only"
+          />
+        </Checkbox>
+      </div>
+      <div className="flex">
+        <FormControlWithLabel
+          label={I18n.t('administration.tokenVoteInstructions')}
+          required
+          type="text"
+          onChange={handleInstructionsChange}
+          value={instructions}
+        />
+        <Helper
+          helperUrl="/static2/img/helpers/helper5.png"
+          helperText={I18n.t('administration.helpers.tokenVoteInstructions')}
+          helperTextClassName="helper-text-only"
+        />
+      </div>
+      <div className="flex">
+        <label htmlFor="input-dropdown-addon">{I18n.t('administration.tokenTypeNumber')}</label>
+        <Helper helperUrl="/static2/img/helpers/helper2.png" helperText={I18n.t('administration.helpers.tokenTypeNumber')} />
+      </div>
+      <DropdownButton
+        title={tokenTypeNumber}
+        onSelect={(e) => {
+          handleTokenVoteTypeNumberChange(e, tokenTypeNumber);
+        }}
+        id="input-dropdown-addon"
+        required
+      >
+        <MenuItem eventKey="1">1</MenuItem>
+        <MenuItem eventKey="2">2</MenuItem>
+        <MenuItem eventKey="3">3</MenuItem>
+        <MenuItem eventKey="4">4</MenuItem>
+      </DropdownButton>
+      {tokenTypeNumber > 0 ? (
+        <div>
+          <div className="separator" />
+          {tokenTypes.map((id, index) => <TokenTypeForm key={`token-type-${index}`} id={id} editLocale={editLocale} />)}
```

Comment:
please avoid index in react keys, here you can use `id` instead (if the order of the list changes, the index will change but you don't want to re-render the `TokenTypeForm` component itself)